### PR TITLE
Bug 1981272: When deleting PVC inside PVC page the status in the heading doesn't match the status field

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
/assign @spadgett

### CODE REVIEW

**Comment by cyril-ui-developer** [comment]
**Before:**
Screen Shot 2021-07-21 at 10 17 31 AM

**After:**
Screen Shot 2021-07-21 at 10 17 10 AM

File: frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ const PVCSummary: React.FC<PVCSummaryProps> = ({ persistentVolumeClaim }) => {
       </dd>
       <dt>{t('console-app~Status')}</dt>
       <dd>
-        <PVCStatus pvc={persistentVolumeClaim} />
+        <PVCStatus {...persistentVolumeClaim} />
```
Comment (spadgett) [inline frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx:108]: Why is this change needed? It seems better to pass the PVC object as a single prop.

File: frontend/public/components/persistent-volume-claim.jsx [hunk 2]
```diff
@@ -54,17 +54,19 @@ const menuActions = [
...
   Edit,
-  (kind, obj) => ({
+  (kind, pvc) => ({
     label: i18next.t('public~Delete PersistentVolumeClaim'),
     callback: () =>
       deletePVCModal({
-        pvc: obj,
+        kind,
+        pvc,
+        redirectTo: `/k8s/ns/${pvc.metadata.namespace}/${PersistentVolumeClaimModel.plural}`,
```
Comment (spadgett) [inline frontend/public/components/persistent-volume-claim.jsx:63]: Why is `redirectTo` needed? It looks like we changed the default behavior to go to the list page now.

Comment (spadgett) [reply frontend/public/components/persistent-volume-claim.jsx:63]: Avoid building paths manually. Use one of the helpers instead.

File: frontend/public/components/persistent-volume-claim.jsx [hunk 3]
```diff
@@ -122,7 +124,7 @@ const PVCTableRow = connect(mapStateToProps)(({ obj, index, key, style, metrics
         <ResourceLink kind="Namespace" name={namespace} title={namespace} />
       </TableData>
       <TableData className={tableColumnClasses[2]}>
-        <PVCStatus pvc={obj} />
+        <PVCStatus {...obj} />
```
Comment (spadgett) [inline frontend/public/components/persistent-volume-claim.jsx:127]: Again, I prefer passing the PVC object as a prop. It's unclear why this was needed.

File: frontend/public/components/persistent-volume-claim.jsx [hunk 4]
```diff
@@ -431,9 +433,11 @@ export const PersistentVolumeClaimsPage = (props) => {
     />
   );
 };
+
 export const PersistentVolumeClaimsDetailsPage = (props) => (
   <DetailsPage
     {...props}
+    getResourceStatus={PVCStatus}
```
Comment (spadgett) [inline frontend/public/components/persistent-volume-claim.jsx:440]: We shouldn't be passing a react component as a value for this prop. `getResourceStatus` should return a string.

**Review by spadgett — CHANGES_REQUESTED** [review]

File: frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx (hunk 1 above)
Comment (cyril-ui-developer) [reply frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx:108]: > Why is this change needed? It seems better to pass the PVC object as a single prop.

@spadgett I changed it because the DetailsPage component  `getResourceStatus` wouldn't accept PVCStatus `pvc` props object -`{pvc:obj}`. It's expecting `{obj}`. 
However since the `getResourceStatus` is expecting string, I will revert it, then find another way to get status for `getResourceStatus`.

File: frontend/public/components/persistent-volume-claim.jsx (hunk 2 above)
Comment (cyril-ui-developer) [reply frontend/public/components/persistent-volume-claim.jsx:63]: > Why is `redirectTo` needed? It looks like we changed the default behavior to go to the list page now.

@spadgett I tested in 4.9, it is not redirecting after delete action.

Comment (cyril-ui-developer) [reply frontend/public/components/persistent-volume-claim.jsx:63]: > Avoid building paths manually. Use one of the helpers instead.

Noted. Thanks

File: frontend/public/components/persistent-volume-claim.jsx (hunk 3 above)
Comment (cyril-ui-developer) [reply frontend/public/components/persistent-volume-claim.jsx:127]: > Again, I prefer passing the PVC object as a prop. It's unclear why this was needed.

Will address.

File: frontend/public/components/persistent-volume-claim.jsx (hunk 4 above)
Comment (cyril-ui-developer) [reply frontend/public/components/persistent-volume-claim.jsx:440]: > We shouldn't be passing a react component as a value for this prop. `getResourceStatus` should return a string.

Will address. Thanks.

File: frontend/public/components/persistent-volume-claim.jsx (hunk 2 above)
Comment (cyril-ui-developer) [reply frontend/public/components/persistent-volume-claim.jsx:63]: > > Why is `redirectTo` needed? It looks like we changed the default behavior to go to the list page now.
> 
> @spadgett I tested in 4.9, it is not redirecting after delete action.

The PVC is not using the common delete modal that has redirect.

**Comment by cyril-ui-developer** [comment]
@spadgett Updated! Thanks.

File: frontend/public/components/modals/delete-pvc-modal.tsx [hunk 5]
```diff
@@ -72,6 +79,8 @@ const DeletePVCModal = withHandlePromise<DeletePVCModalProps>((props) => {
 
 export type DeletePVCModalProps = {
   pvc: PersistentVolumeClaimKind;
+  redirectTo?: string;
```
Comment (spadgett) [inline frontend/public/components/modals/delete-pvc-modal.tsx:82]: It looks like this prop is no longer used and can be removed from the types?

**Review by spadgett — COMMENTED** [review]
Thanks @cyril-ui-developer, this looks cleaner

File: frontend/public/components/modals/delete-pvc-modal.tsx (hunk 5 above)
Comment (cyril-ui-developer) [reply frontend/public/components/modals/delete-pvc-modal.tsx]: > It looks like this prop is no longer used and can be removed from the types?

@spadgett Updated.

File: frontend/public/components/modals/delete-pvc-modal.tsx [hunk 6]
```diff
@@ -72,6 +79,7 @@ const DeletePVCModal = withHandlePromise<DeletePVCModalProps>((props) => {
 
 export type DeletePVCModalProps = {
   pvc: PersistentVolumeClaimKind;
+  kind: K8sKind;
```
Comment (spadgett) [inline frontend/public/components/modals/delete-pvc-modal.tsx:82]: This should always be `PersistentVolumeClaimModel` right? Do we need to pass `kind` as a prop?

Comment (cyril-ui-developer) [reply frontend/public/components/modals/delete-pvc-modal.tsx:82]: > This should always be `PersistentVolumeClaimModel` right? Do we need to pass `kind` as a prop?

Yes, it will. Oh I see! No. 
Will use 'PersistentVolumeClaimModel` instead of passing `kind` for `resourceListPathFromModel` method. 
Thanks for the catch :)

Comment (cyril-ui-developer) [reply frontend/public/components/modals/delete-pvc-modal.tsx]: > This should always be `PersistentVolumeClaimModel` right? Do we need to pass `kind` as a prop?

Updated.

**Review by spadgett — APPROVED** [review]
/lgtm

Thanks @cyril-ui-developer 👍
